node/src/actor_system_factory.rs

```diff
@@ -1491,6 +1464,57 @@ mod tests {
         // no not-enough-results-provided error: test passes
     }
 
+    #[test]
+    fn start_automap_change_handler_handles_ip_changes_properly() {
+        let mut subject = ActorSystemFactoryToolsReal::new();
+        let make_params_arc = Arc::new(Mutex::new(vec![]));
+        let automap_control = AutomapControlMock::new()
+            .get_mapping_protocol_result(None)
+            .get_public_ip_result(Ok(IpAddr::from_str("1.2.3.4").unwrap()))
+            .add_mapping_result(Ok(()));
+        subject.automap_control_factory = Box::new(
+            AutomapControlFactoryMock::new()
+                .make_params(&make_params_arc)
+                .make_result(Box::new(automap_control)),
+        );
+        let mut config = BootstrapperConfig::default();
+        config.mapping_protocol_opt = None;
+        config.neighborhood_config.mode = NeighborhoodMode::Standard(
+            NodeAddr::new(&IpAddr::from_str("0.0.0.0").unwrap(), &[1234]),
+            vec![],
+            DEFAULT_RATE_PACK,
+        );
+        let (recorder, _, recording_arc) = make_recorder();
+        let new_ip_recipient = recorder.start().recipient();
+
+        subject.start_automap(
+            &config,
+            Box::new(PersistentConfigurationMock::new()),
+            vec![new_ip_recipient],
+        );
+
+        let make_params = make_params_arc.lock().unwrap();
+        assert_eq!(make_params[0].0, None);
```

Comment:
Giving a variable name to `make_params[0].0` might be helpful for easier going through the test.

Reply:
Done.